docs/dev/Lua API.rst

```diff
@@ -5130,6 +5130,17 @@ widget does not require direct usage of ``Tab``.
            usage of ``Tab`` in ``TabBar:init()`` for an example. See the default value of ``active_tab_pens`` or ``inactive_tab_pens``
            in ``TabBar`` for an example of how to construct pens.
 
+RangeSlider class
+-----------------
+
+This widget implements a mouse-interactable range-slider. The user can move its two handles to set minimum and maximum values to define a range.
+
+:num_stops: Specifies the amount of "notches" in the range slider, the places where handles can stop.
+:get_left_idx_fn: The function used by the RangeSlider to determine what value to display on its left handle.
+:get_right_idx_fn: The function used by the RangeSlider to determine what value to display on its right handle.
```

Comment:
it's more to get the index of the stop to display the handle at -- we don't actually display the value. the purpose of these callbacks is to allow the parent to keep the slider value state and thus give the parent control over the values (allowing them to be changed by hotkeys, for example). I did it this way to give flexibility to the layout. If this class kept the slider position state internally, we'd also need to provide options for how to display the hotkey hint widgets that allow you to change the values with the keyboard.

Reply:
It might be worth mentioning that the parent controls the index state in the introduction paragraph.

Reply:
So hold on... does the rangeslider always need a parent CycleHotkeyLabel?
Why can we define the number of stops then? 🤔 shouldn't the slider just mirror the number of available options from the parent?

i ran into this weirdness while attempting to write the single-handle slider class. I could set the slider to have less notches than the parent CycleHotkeyLabel, but then the handle would go _past_ the end anyway. <_<